### DISCUSSION TITLE
unable to deploy on zksync even if my fundme.sol file is in "contracts" folder

### DISCUSSION THREAD

**Discussion by satyaorz** [Cyfrin/foundry-full-course-cu#2821] [Q&A] [answered]
i am able to compile the file but unable to deploy it as it says
"No contracts ready for deployment yet, compile a solidity contract"

![Screenshot From 2024-11-07 02-15-45](https://github.com/user-attachments/assets/cc71188c-4b11-4b49-b696-6d2d0f205225)
![Screenshot From 2024-11-07 02-15-56](https://github.com/user-attachments/assets/17b77caf-17ff-454a-8abd-8b40e8e30eff)

**Comment by EngrPips** [comment]
I will say refresh your browser and retry.

**Reply by satyaorz** [reply]
no its not working this time, i tried different compiler version also and without the "contract" folder. same issue.

okay now i tried to compile the storagefactory.sol after moving it to the "contracts" folder and it was showing in the deploy section. but its not working with fundme.sol !?

**Reply by satyaorz** [reply]
okay so i fixed it. the issue was the use of library, even though the library function was in the same "fundme.sol" file, it was causing the issue i guess?? so i removed the library and defined all the functions that were in the ```library PriceConverter``` inside the ```contract Fundme {}```. and then when i compiled it, it was showing the contract in deploy section.

**Reply by EngrPips** [reply]
Hello @butterpaneermasala, how do you mean `removing the library and using the functions instead`

**Reply by satyaorz** [reply, marked as answer]
oh what i mean is,
```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.24;

import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";

library PriceConverter { //this was the library
    function GetPrice() public view returns(uint256) {
        AggregatorV3Interface PriceFeed = AggregatorV3Interface(0xfEefF7c3fB57d18C5C6Cdd71e45D2D0b4F9377bF);
        (, int256 price, , ,) = PriceFeed.latestRoundData();
        return uint256(price * 1e10);
    }

    function GetConversionRate(uint256 ethAmount) public view returns(uint256) {
        uint256 ethPrice = GetPrice();
        uint256 ethPriceinUSD = (ethPrice * ethAmount) / 1e18;
        return ethPriceinUSD;
    }

    function GetVersion() public view returns(uint256) {
        return AggregatorV3Interface(0xfEefF7c3fB57d18C5C6Cdd71e45D2D0b4F9377bF).version();
    }
}

error NOTowner();

contract Fundme {

    address public immutable i_owner;
```
this is how i was using the library, like how patrick did.

but then i removed the library and used the functions of the library by defining them inside the ```contract fundme(){}```
like this
```solidity
contract Fundme {

    address public immutable i_owner;

    constructor() {
        i_owner = msg.sender;
    }
    // and now here is the functions that were inside the library
    function GetPrice() public view returns(uint256) {
        AggregatorV3Interface PriceFeed = AggregatorV3Interface(0xfEefF7c3fB57d18C5C6Cdd71e45D2D0b4F9377bF);
        (, int256 price, , ,) = PriceFeed.latestRoundData();
        return uint256(price * 1e10);
    }

    function GetConversionRate(uint256 ethAmount) public view returns(uint256) {
        uint256 ethPrice = GetPrice();
        uint256 ethPriceinUSD = (ethPrice * ethAmount) / 1e18;
        return ethPriceinUSD;
    }

    function GetVersion() public view returns(uint256) {
        return AggregatorV3Interface(0xfEefF7c3fB57d18C5C6Cdd71e45D2D0b4F9377bF).version();
    }
    uint256 public constant minimumUSD = 5 * 1e18;
```
and then it worked.

**Reply by EngrPips** [reply]
Oh I see, That makes sense.

**Comment by satyalohit** [comment]
Hello, even I faced this issue thank you for explaining the solution @butterpaneermasala .

**Reply by EngrPips** [reply]
Interesting.
